src/Compilers/Core/Portable/Compilation/DeterministicKeyBuilder.cs

```diff
@@ -191,6 +200,44 @@ void writeGenerators()
                 }
                 writer.WriteArrayEnd();
             }
+
+            void writeResources()
+            {
+                writer.WriteArrayStart();
+                foreach (ResourceDescription resource in resources)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+                    writer.WriteObjectStart();
+                    writer.Write("resourceName", resource.ResourceName);
+                    writer.Write("fileName", resource.FileName);
+                    writer.Write("isPublic", resource.IsPublic);
+                    writer.WriteKey("content");
+                    writeResourceContent(writer, resource);
+                    writer.WriteObjectEnd();
+                }
+                writer.WriteArrayEnd();
+            }
+
+            static void writeResourceContent(JsonWriter writer, ResourceDescription resource)
+            {
+                if (resource.IsEmbedded)
+                {
+                    writer.WriteObjectStart();
+                    WriteByteArrayValue(writer, "checksum", computeChecksum(resource));
+                    writer.WriteObjectEnd();
+                }
+                else
+                {
+                    writer.WriteNull();
+                }
+            }
+
+            static Span<byte> computeChecksum(ResourceDescription resource)
+            {
+                using var stream = resource.DataProvider();
+                using var hashAlgorithm = System.Security.Cryptography.SHA256.Create();
```

Comment:
:memo: `sourceText.GetChecksum` (in `WriteSourceText`) uses either SHA1 or SHA256. I figured it's okay to align with that for now, but we should consider moving to XXHASH at some point

Reply:
The key file should use the same checksum for all entries. This is effectively a merkle tree and the checksums should be uniform throughout it. This guarantees that the file has the same efficacy irrespective of what compiler options are used. 

> we should consider moving to XXHASH at some point

Disagree. The intent of these checksums is to provide uniqueness and that is one of the valid uses of crypto. The code here should just use the default crypto algorithm for all entries.